File: Go/proj_euler_problem_0010.go
/*
 * Problem 10
 * Summation of primes
 *
 * The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
 * 
 * Find the sum of all the primes below two million.
 *
 */

package main

import (
    "fmt"
    "math"
)

func isPrime(a uint64) bool {
    var a_float float64 = float64(a)
    var b uint64 = uint64(math.Sqrt(a_float))
    var i uint64 = 0
    for i = 2; i <= b; i++ {
        if a % i == 0 {
            return false
        }
    }
    return true
}

func main() {
    var a uint64 = 2000000
    var sum uint64 = 0
    var i uint64 = 1
    for i = 2; i <= a; i++ {
        if isPrime(i) {
            sum += i
        }
    }
    fmt.Println(sum)
}    

File: Python/proj_euler_problem_0010.py
'''
Problem 10
Summation of primes

The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.
'''
import math
import sys


def findPrimes():
    input = 2000000
    primes = [2]
    
    primeFlag = 1
    for i in range (3, input + 1, 2):                
        for j in range (2, int(math.sqrt(i)) + 1):
            if i % j == 0:
                primeFlag = 0
                break
        if primeFlag:
            primes.append(i)
        else:
            primeFlag = 1
    sum = 0
    for i in primes:
        sum += i
    
    print(sum)

            
def EulerProblem0010():
    findPrimes()
        
if __name__ == "__main__":
    EulerProblem0010()
    